stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -1743,17 +1832,36 @@ impl NakamotoChainState {
             return Ok(false);
         };
 
-        let schnorr_signature = &block.header.signer_signature.0;
-        if !db_handle.expects_signer_signature(
-            &block.header.consensus_hash,
-            schnorr_signature,
-            &block.header.signer_signature_hash().0,
-            aggregate_public_key,
-        )? {
-            let msg = format!(
-                "Received block, but the signer signature does not match the active stacking cycle"
+        // TODO: epoch gate to verify aggregate signature
+        // let schnorr_signature = &block.header.signer_signature.0;
+        // if !db_handle.expects_signer_signature(
+        //     &block.header.consensus_hash,
+        //     schnorr_signature,
+        //     &block.header.signer_signature_hash().0,
+        //     aggregate_public_key,
+        // )? {
+        //     let msg = format!(
+        //         "Received block, but the signer signature does not match the active stacking cycle"
+        //     );
+        //     warn!("{}", msg; "aggregate_key" => %aggregate_public_key);
+        //     return Err(ChainstateError::InvalidStacksBlock(msg));
+        // }
+
+        // TODO: epoch gate to verify signatures vec
+        if let Err(e) = block.header.verify_signer_signatures(&reward_set) {
+            let reward_set_keys = reward_set
+                .clone()
+                .signers
+                .unwrap()
```

Comment:
We try to avoid using `.unwrap()` in consensus-critical code.  Also, this `.unwrap()` is only performed in the service of logging some debugging information, so it should really be removed outright.

Reply:
I went ahead and removed the whole `reward_set_keys` variable and changed this to just be a `warn!` and returning that error